frontend/app/components/UI/Button.tsx

```diff
@@ -1,6 +1,13 @@
-import { TextButton } from "../../types/buttons";
+import { ButtonHTMLAttributes, forwardRef } from "react";
 
-const Button = ({ label, variant = "filled", ...rest }: TextButton) => {
+interface TextButton extends ButtonHTMLAttributes<HTMLButtonElement> {
+  variant?: "filled" | "outlined";
+}
+
+const Button = forwardRef<HTMLButtonElement, TextButton>(function Button(
+  { variant = "filled", children, ...rest },
+  ref,
+) {
   const variants: { [key: string]: string } = {
```

Comment:
Kan vi dra ut `variant` typen sånn at det ikke går an å be om en som ikke finnes? 😄 
```suggestion
  const variants: Record<ButtonVariant, string> = {
```